dataeng/jobs/AnalyticsEmailOptinMaster.groovy

```diff
@@ -0,0 +1,106 @@
+        stringParam('OUTPUT_PREFIX','email-opt-in-', 'Optional prefix to prepend to output filename.')
+        stringParam('NOTIFICATION_EMAILS','analytics@edx.opsgenie.net analytics-alerts@edx.org',
+                'Space separated list of emails to notify in case of failure.')
+        stringParam('DATE_MODIFIER','',
+                'Used to set the date of the CWSM dump.  Leave blank to use today\'s date.  Set to "-d 201x-0x-0x" if that is when the CWSM dump took place, typically the preceding Sunday.  (Leave off quotes.)')
+        stringParam('ORG_CONFIG','data-czar-keys/config.yaml', 'Path to the data-czar organization config file')
+        stringParam('DATA_CZAR_KEYS_BRANCH','master', 'Branch of the Data-czar-keys repository to use')
+    }
+
+    multiscm{
+        git {
+            remote {
+                url('git@github.com:edx/edx-platform.git')
+                branch('$PLATFORM_BRANCH')
+                credentials('1')
+            }
+            extensions {
+                pruneBranches()
+                relativeTargetDirectory('edx-platform')
+            }
+        }
+        git {
+            remote {
+                url('git@github.com:edx/edx-analytics-exporter.git')
+                branch('$EXPORTER_BRANCH')
+                credentials('1')
+            }
+            extensions {
+                pruneBranches()
+                relativeTargetDirectory('analytics-exporter')
+            }
+        }
+        git {
+            remote {
+                url('git@github.com:edx-ops/analytics-secure.git')
+                branch('$SECURE_BRANCH')
+                credentials('1')
+            }
+            extensions {
+                pruneBranches()
+                relativeTargetDirectory('analytics-secure')
+            }
+        }
+        git {
+            remote {
+                url('git@github.com:edx/data-czar-keys.git')
+                branch('$DATA_CZAR_KEYS_BRANCH')
+                credentials('1')
+            }
+            extensions {
+                relativeTargetDirectory('data-czar-keys')
+            }
+        }
+    }
+
+    wrappers {
+        timestamps()
+    }
+
+    steps {
+        virtualenv {
+            nature("shell")
+            command(
+                readFileFromWorkspace("dataeng/resources/setup-platform-venv.sh")
+            )
+        }
+        virtualenv {
+            nature("shell")
+            command(
+                readFileFromWorkspace("dataeng/resources/setup-exporter-properties.sh")
+            )
+        }
+        downstreamParameterized {
+            trigger('analytics-email-optin-worker') {
+                block {
+                    buildStepFailure('FAILURE')
+                    failure('FAILURE')
+                    unstable('UNSTABLE')
+                }
+                parameters {
+                    predefinedProp('MASTER_WORKSPACE', '${WORKSPACE}')
```

Comment:
Should these be removed as these are also declared on the worker ?

Reply:
Great question.  I suspect yes because the shell code is rolling them into the environment variables.  But also let's run a confirmation run after these have been removed.